clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -625,6 +625,7 @@ private void completeBatch(ProducerBatch batch, ProduceResponse.PartitionRespons
             if (transactionManager != null)
                 transactionManager.removeInFlightBatch(batch);
             this.accumulator.splitAndReenqueue(batch);
+            maybeRemoveFromInflightBatches(batch);
             this.accumulator.deallocate(batch);
```

Comment:
Would it make sense to have a method that deallocates and removes from in flight batches?

Reply:
Yeah, I think that's reasonable. Added.